README.md

```diff
@@ -385,6 +399,126 @@ jsn:select({value, [<<"user">>, <<"id">>]},
 % [<<"456">>]
 ```
 
+#### <a name="with-without"/>Using `with/2` and `without/2`
+
+The functions `with/2` and `without/3` accept a list of `path()`s or a
+`path_elements_map()` defined in [jsn.hrl](include/jsn.hrl). These functions
+are inspired by [`maps:with/2`](https://erlang.org/doc/man/maps.html#with-2) and
+[`maps:without/2`](https://erlang.org/doc/man/maps.html#without-2), but support
+the nested path formats of this library. Essentially, you can pass a list of
+arbitrarily nested paths to either function, and they will return your input
+with those paths applied, either by including **only** those paths in the input
+in the case of `with/2`, or removing those paths from the input in the case of
+`without/2`.
+
+However, because nested paths are a little more complex than working with the
+flat keys list of the `maps` analogues of these functions, there are some
+caveats and specific aspects to be aware of when using these functions.
+
+* When using `with/2` or `without/2` for sets of objects, or reusing it
+  regularly, it is recommended to 'compile' your `Paths` input using
+  `jsn:path_elements_map(Paths)`, and pass the resulting map directly to the
+  functions. If this isn't done, the `Paths` will be parsed into this structure
+  on every iteration.
+* **Use caution with atom keys in objects**. These functions use the function
+  `jsn:path_elements_map(Paths)` to construct a structure for the nested
+  traversals of the object, and for overall performance, it checks for the
+  possibility that any individual key could be an atom when constructing that
+  structure. That means that if you call this function before constructing your
+  object, and you use new atoms when doing so, those atom-keyed path elements
+  may not be included (or excluded) as expected. If your objects have atoms,
+  it is recommended to ensure all atoms are referenced in the system (i.e., in
```

Comment:
Is it actually feasible to check if an atom is in the atom table? I feel like like the recommendation here is better left out.

Reply:
Yes, `binary_to_existing_atom/2` is how we check. As for ensuring that they're present, you just have to reference them in compiled code. This is not better left out IMO.